3_extract/src/process_glm_output.R

```diff
@@ -1,30 +1,26 @@
 #' @title Combine output from GLM model runs
-#' @description function to read in the raw output from the GLM
-#' runs for each lake-gcm combo (set by grouping of 
-#' `p2_glm_uncalibrated_run_groups`) for each time period, filter 
-#' that output to the valid dates within each time period 
-#' (excluding the burn-in and burn-out periods), and save the
-#' result as a single feather file
-#' @param run_groups a grouped version of the `p2_glm_uncalibrated_runs`
-#' output tibble subset to the site_id, gcm, time_period, raw_meteo_fl, 
-#' export_fl, and export_fl_hash columns and grouped by site_id and gcm.
-#' Then filtered to only groups for which glm_success==TRUE for all runs
-#' in that group. The function maps over these groups.
-#' @param lake_cell_tile_xwalk - mapping of which lakes fall into which 
-#' gcm cells and tiles
+#' @description function to read in the raw output from the 3 GLM
+#' runs for each fully successful lake-gcm combo (set by grouping of 
+#' `p2_glm_uncalibrated_run_groups`), filter that output to the 
+#' burn-out periods, and save the result as a single feather file
+#' @param run_group a single group from the `p2_glm_uncalibrated_run_groups`
+#' grouped version of the `p2_glm_uncalibrated_runs` output tibble subset 
+#' to the site_id, gcm, time_period, raw_meteo_fl, export_fl, and 
+#' export_fl_hash columns and grouped by site_id and gcm. Then filtered to 
+#' only groups for which glm_success==TRUE for all runs in that group. 
+#' The function maps over these groups.
 #' @param outfile_template the template for the name of the
 #' final output feather file
-#' @return a tibble with one row per lake-gcm combo which includes the 
-#' site_id, gcm, the name of the export feather file, its hash, the state 
-#' the lake is in, and the cell_no and tile_no for the GCM data for that lake. 
-combine_glm_output <- function(run_groups, lake_cell_tile_xwalk, outfile_template) {
+#' @return a single feather file with the output data for that lake-gcm
+#' combo, filtered to the valid dates from each time period
+combine_glm_output <- function(run_group, outfile_template) {
   # set filename
-  outfile <- sprintf(outfile_template, unique(run_groups$site_id), unique(run_groups$gcm))
+  outfile <- sprintf(outfile_template, unique(run_group$site_id), unique(run_group$gcm))
   
   # combine into single feather file and write
   # truncating output for each time period to valid dates
   # (excluding burn-in and burn-out periods)
-  purrr::map2_df(run_groups$raw_meteo_fl, run_groups$export_fl, function(raw_meteo_fl, export_file) {
+  purrr::map2_df(run_group$raw_meteo_fl, run_group$export_fl, function(raw_meteo_fl, export_file) {
```

Comment:
Stating this explicitly as part of my review in case there is a future question, even though we passed in a tibble with these filenames as the dependency of this function/target, this `combine_glm_output()` will still be triggered to build if the contents of `export_fl` change because the `run_group` tibble will have a different value in the `export_fl_hash` column if the data file was different.